Structural episode (#55)
4TUResearchData-Carpentries/interoperability-climate-sciences: episodes/semantic-interoperability.md

---
title: "Semantic interoperability"
teaching: 30
exercises: 20
---

:::::::::::::::::::::::::::::::::::::: questions

> * What is semantic interoperability, and which kinds of meaning must be made explicit?

Reviewer: ```suggestion
* What is semantic interoperability?
* What types of meaning must be made explicit to enable semantic interoperability?
```
I would split up the first question into two to make if clearer

* Use the [Annotator](https://earthportal.eu/annotator) to identify ontology concepts that may describe terms occurring in documentation or metadata.
* Inspect mappings, identifiers, classes, properties, provenance, submissions, and available machine-readable representations.
* After creating an account and signing in, use **Submit ontology** to share an ontology or another semantic artefact with the wider Earth-science community.

Publishing a semantic artefact in a catalogue can improve its visibility and reuse, but catalogue inclusion is not by itself evidence that the resource is authoritative, current, adequately licensed, or appropriate for a particular dataset. Before adopting a term or identifier, inspect:

* who created and maintains the resource;
* its scope and intended use;
* licence and access conditions;
* version and modification history;
* persistent namespace or concept identifiers;
* governance and term-submission process;
* mappings to other vocabularies; and
* whether the relevant scientific community actively uses it.

#### Explore further: the Climate and Forecast ontology representation

> EarthPortal includes the [Climate and Forecast (CF) features ontology](https://earthportal.eu/ontologies/CFF), an OWL representation of generic features derived from the CF Standard Names vocabulary. It exposes CF-related concepts through ontology classes, properties, individuals, identifiers, and relationships that can be explored programmatically or through the portal interface.

Reviewer: It's good to explain what RDF/OWL is and use the full name together with the abbrevation:  "Web Ontology Language (OWL)". The explanation can be included in the glossary/ references and the reference linked here.  There is a mention of RDF/OWL before as an example, but its meaning is not provided either, so I think it might be confusing rather than helpful.

When exploring it, compare the portal entry with the authoritative CF resources and ask:

* Does the concept correspond to a current CF standard name?
* Is its definition identical to, derived from, or older than the current CF definition?
* Is the concept represented as a class, property, or individual?
* Which relationships have been made explicit in OWL?
* Are provenance, licence, maintenance responsibility, and update frequency sufficiently clear?
* Would referencing this ontology URI improve machine-actionable meaning in the intended workflow, or is the canonical CF term and definition the more appropriate reference?

A useful search exercise is to look for terms such as `air_temperature`, `precipitation_flux`, or radar-related quantities and compare what EarthPortal exposes with the current CF Standard Name Table.

:::::::::::::::::::::::::::::::::::::

## Semantic interoperability is not provided by a file format

> NetCDF, Zarr, CSV, TSV, Parquet, GeoTIFF, and other formats can all carry data that are semantically clear—or semantically ambiguous.

Reviewer: ```suggestion
NetCDF, Zarr, CSV, TSV, Parquet, GeoTIFF, and other formats can all carry data that are semantically clear - or semantically ambiguous.
```

flag_masks
flag_meanings
```

Without these relationships and code definitions, a value such as `2` in a quality column has no stable machine-actionable meaning.

### Feature types and observation geometry

For time series, trajectories, profiles, and other discrete sampling geometries, CF can declare a `featureType` and prescribe how observations, stations, trajectories, profiles, and coordinates relate.

This allows tools to distinguish, for example, a station time series from a gridded field even when both are stored in NetCDF.

## CF is a convention, not a complete description of every research context

CF is powerful, but CF compliance does not guarantee:

* that the scientific values are correct;
* that calibration or processing was appropriate;
* that uncertainty is adequately described;
* that all provenance is available;
* that discovery metadata are complete;
* that two datasets use the same spatial or temporal resolution;
* that two variables are suitable for the same research question;
* that missing values are acceptably limited; or
* that a dataset is free from software or production errors.

Other standards and metadata profiles may complement CF. For example:

> * dataset-discovery metadata may be expressed through ACDD, ISO 19115, DataCite, or repository metadata;

Reviewer: I think it's good to explain those: ACDD, ISO 19115, DataCite - I don't see any mention of them in other parts of the workshop. 
 ( possibly in the glossary, with a link)
